File: resolutions/src/utils/isSolidity.js
/**
 * Util to determine if a file path is a solidity file
 * @param {string} fileName - The file name to check
 * @returns {boolean} - Whether or not the file is a solidity file
 * @example
 * ```ts
 * const isSolidity = isSolidity('Contract.sol')
 * console.log(isSolidity) // true
 * ```
 */
export const isSolidity = (fileName) =>
	fileName.endsWith('.sol') &&
	!fileName.endsWith('/.sol') &&
	fileName !== '.sol'

File: resolutions-go/internal/utils/is_solidity.go
package utils

import (
	"strings"
)

// IsSolidity determines if a file path is a Solidity file
func IsSolidity(fileName string) bool {
	return strings.HasSuffix(fileName, ".sol") && !strings.HasSuffix(fileName, "/.sol") && fileName != ".sol"
}